posthog/hogql_queries/web_analytics/web_overview_pre_aggregated.py

```diff
@@ -20,41 +20,94 @@ def __init__(self, runner: "WebOverviewQueryRunner") -> None:
+            """
+            SELECT
+                {unique_persons_current} AS unique_persons,
+                {unique_persons_previous} AS previous_unique_persons,
+
+                {pageviews_current} AS pageviews,
+                {pageviews_previous} AS previous_pageviews,
+
+                {unique_sessions_current} AS unique_sessions,
+                {unique_sessions_previous} AS previous_unique_sessions,
+
+                {avg_session_duration_current} AS avg_session_duration,
+                {avg_session_duration_previous} AS previous_avg_session_duration,
+
+                {bounce_rate_current} AS bounce_rate,
+                {bounce_rate_previous} AS previous_bounce_rate,
+
+                NULL AS revenue,
+                NULL AS previous_revenue
+            FROM web_overview_daily
+            """,
+            placeholders={
+                "unique_persons_current": self._uniq_merge_if("persons_uniq_state", current_period_filter),
+                "unique_persons_previous": self._uniq_merge_if("persons_uniq_state", previous_period_filter),
+                "pageviews_current": self._sum_merge_if("pageviews_count_state", current_period_filter),
+                "pageviews_previous": self._sum_merge_if("pageviews_count_state", previous_period_filter),
+                "unique_sessions_current": self._uniq_merge_if("sessions_uniq_state", current_period_filter),
+                "unique_sessions_previous": self._uniq_merge_if("sessions_uniq_state", previous_period_filter),
+                "avg_session_duration_current": self._safe_avg_sessions(
+                    "total_session_duration_state", current_period_filter
+                ),
+                "avg_session_duration_previous": self._safe_avg_sessions(
+                    "total_session_duration_state", previous_period_filter
+                ),
+                "bounce_rate_current": self._safe_avg_sessions("total_bounces_state", current_period_filter),
+                "bounce_rate_previous": self._safe_avg_sessions("total_bounces_state", previous_period_filter),
+            },
+        )
+
+        assert isinstance(query, ast.SelectQuery)
 
         filters = self._get_filters(table_name="web_overview_daily")
         if filters:
             query.where = filters
 
         return query
+
+    def _uniq_merge_if(self, state_field: str, period_filter: ast.Expr) -> ast.Call:
+        """Utility method to create uniqMergeIf expressions"""
+        return ast.Call(
+            name="uniqMergeIf",
+            args=[
+                ast.Field(chain=[state_field]),
+                period_filter,
+            ],
+        )
+
+    def _sum_merge_if(self, state_field: str, period_filter: ast.Expr) -> ast.Call:
+        """Utility method to create sumMergeIf expressions"""
+        return ast.Call(
+            name="sumMergeIf",
+            args=[
+                ast.Field(chain=[state_field]),
+                period_filter,
+            ],
+        )
```

Comment:
Those are somewhat useless right now, but I plan for them to be helpful when using the state combinators in a future PR, it is just less changes if they're on this one.